### DISCUSSION TITLE
Custom role - Navigation links

### DISCUSSION THREAD

**Discussion by applicationuser** [Budibase/budibase#7849] [Help]
Hello ,

Need help with navigation link setup with custom roles.

anything i missed or setup incorrectly? Please help

Here is my setup

Custom role: Regional inherits Basic Role
 
![image](https://user-images.githubusercontent.com/42922833/191123951-fdc4ba85-502d-487c-bbb2-650b2b16b48a.png)

Navigation links - configured Minimum role to access link as Basic.

![image](https://user-images.githubusercontent.com/42922833/191124175-fccf0433-b32f-4794-a8a0-11d78a203031.png)

user setup with app access set to custom role: Regional
![image](https://user-images.githubusercontent.com/42922833/191125168-4599dcf8-ff3c-4a6a-b3f3-dbf0c8066a79.png)

Published app and logged in with user defined at role: Regional 
Navigation links not displayed on Home page 

![image](https://user-images.githubusercontent.com/42922833/191125562-851f7cc5-c228-4839-8365-618f5ef3a180.png)

another issue, with the dev mode preview there is no option to view screen as custom role user 
![image](https://user-images.githubusercontent.com/42922833/191124299-8d35c8b6-12dc-4e71-9f68-0c7dac8a5b66.png)

**Comment by kasgithub8** [comment]
I think for the not showing part you have to change it from basic to your custom role.
2nd is clearly a Budibase problem!

**Reply by applicationuser** [reply]
Thank @kasgithub8 
Basic user role also needs set of navigation links common to custom role.
If we change links minimum role from Basic to custom role then we see duplicate links .

may be i misunderstood navigation link Minimum role ?  if we define custom role extending Basic role all the links defined for Basic are available for custom role.  is that not correct ? 
![image](https://user-images.githubusercontent.com/42922833/191263517-ada3473e-adb1-4b74-b4d7-be1a45757598.png)

**Reply by kasgithub8** [reply]
I just think this whole Custom Role thing is not 100% correct, I had this problem too.

**Reply by pbarmak** [reply]
@applicationuser, I think you are correct in your assumption. I'm having the same issue. I created a custom role that inherits from Basic. I assigned a user to that role. That user cannot see navigation links that have a minimum role of Basic, and they should be able to see that (since the custom role inherits from Basic). I also have links that only the custom role or above should see, but Basic users see that link. I think something is definitely not working correctly with Navigation Links and custom roles, or I'm also misunderstanding how this works. Have you opened an Issue for this yet? I'd like to subscribe to it if you have; if not, I'm happy to open one if you like.

**Reply by applicationuser** [reply]
Thanks @pbarmak  @kasgithub8 

Issue created : https://github.com/Budibase/budibase/issues/7917
